lldb/test/API/python_api/process/io/TestProcessIO.py

```diff
@@ -99,31 +99,45 @@ def test_stdout_stderr_redirection(self):
     @expectedFlakeyLinux(bugnumber="llvm.org/pr26437")
     @skipIfDarwinEmbedded  # debugserver can't create/write files on the device
     def test_stdout_stderr_redirection_to_existing_files(self):
-        """Exercise SBLaunchInfo::AddOpenFileAction() for STDOUT and STDERR without redirecting STDIN to output files already exist."""
+        """Exercise SBLaunchInfo::AddOpenFileAction() for STDOUT and STDERR redirect to output files already exist."""
         self.setup_test()
         self.build()
         self.create_target()
-        self.write_file_with_placeholder(self.output_file)
-        self.write_file_with_placeholder(self.error_file)
-        self.redirect_stdout()
-        self.redirect_stderr()
-        self.run_process(True)
-        output = self.read_output_file_and_delete()
-        error = self.read_error_file_and_delete()
-        self.check_process_output(output, error)
 
-    def write_file_with_placeholder(self, target_file):
+        # Create the output and error files with placeholder
         placeholder = "This content should be overwritten."
         if lldb.remote_platform:
+            f = open(self.local_output_file, "w")
```

Comment:
Why does this create files on your local filesystem now instead of doing the previous `platform file write` commands? Was there something wrong with the previous approach?

Reply:
If we are going to make local files the [Save core tests put them in the build directory](https://github.com/llvm/llvm-project/blob/8a0cb9ac869334fd6c6bd6aad8408623a7ccd7f6/lldb/test/API/functionalities/process_save_core_minidump/TestProcessSaveCoreMinidump.py#L414) as an example.

Reply:
So I double checked the `platform file write` it will require `platform file open` first and obtain the file descriptor so that we can write. 

The put method however won't require extra steps and is used in the same test already.